modelci/app/v1/endpoints/model.py

```diff
@@ -48,16 +49,81 @@ async def get_model_structure(id: str):  # noqa
 
 
 @router.patch('/structure/{id}')  # TODO: add response_model
-def update_model_structure_as_new(id: str, dry_run: bool = False):  # noqa
+def update_model_structure_as_new(id: str, structure, dry_run: bool = False):  # noqa
     """
-    Update model structure and save as a new version.
+    TODO: Update model structure by adjusting layers (add, modify, delete) or rewiring the
+        connections between layers.
+
+    Examples:
+        Fine-tune the model by modify the layer with name 'fc' (last layer). The layer
+        has a changed argument out_features = 10. _op='M' indicates the operation to this layer ('fc')
+        is 'Modify'. There is no changes in layer connections.
+        Therefore, the structure change summary is
+            [M] fc: (...) out_features=10
+
+        >>> structure = {'layer': {'fc': {'out_features': 10, '_op': 'M'}}}
+        ... update_model_structure_as_new(id=..., structure=structure)
+
+        Use original model as a feature extractor. The new model delete the last layer named 'fc', and add two
+        layers as following:
+            fc1: (nn.Linear) in_features=1024, out_features=512
+            fc2: (nn.Linear) in_features=512, out_features=10
+        The node change summary is
+            [D] fc
+            [A] fc1: (nn.Linear) in_features=1024, out_features=512
+            [A] fc2: (nn.Linear) in_features=512, out_features=10
+        Besides, we have connection changes:
+            [D] conv1 -> fc
+            [A] conv1 -> fc1
+            [A] fc1 -> fc2
+
+        >>>
+        ... structure = {
```

Comment:
How about change this structure as 
```
'layer': [
{ layer_name: 'fc', '_op': 'D' },
{ layer_name: 'fc1', '_op': 'A' , 'in_features': 1024, 'out_features': 512, '_type': 'nn.Linear' }
],
'connection': [
{'in_layer': 'fc', 'out_layer': 'fc1', '_op': 'A'}
]
```

Reply:
This increases redundancy. What is the benefit if we switch to such a structure?

Reply:
> This increases redundancy. What is the benefit if we switch to such a structure?

Yeah, it's maybe more complicated. How about implement a sorted dict data strcuture as https://github.com/martinjm97/ENNUI/blob/49beaee37579fd094b11f87cad6d86a6c549105b/src/ui/shapes/layer.ts#L25, we only return an ordered dict for each model, and recreate the whole model based on that as modifying old model can be a tricky job.

Reply:
1. Yes I agree with the idea of using an ordered dict for the `layer` part of the structure.

2. In my view, recreate the whole model based on the new structure is more complicated. 
ENNUI (the link you provided) does not support training from a pre-trained model.  Therefore, when there is an update, it retrains from scratch.
We are supposed to support start from a pre-trained one. We need to compare the new full version and the old full version, see which layers get updated, and retain those layers that are not changed. Let the frontend to keep track of the changed layers by the user is a much easier job.